### PR TITLE
Bobp improvement

### PR DESCRIPTION
**Pull request by idruzhitskiy**

### CODE REVIEW

**Comment by agend** [comment]
LGTM. @pyakushin  take a look

File: src/bin/bobp.rs [hunk 1]
```diff
@@ -1,150 +1,494 @@
...
+            put_error,
+            get_error,
+            put_count_spd as f64 * k,
+            get_count_spd as f64 * k,
+            finite_or_default(cur_st_put_time / cur_st_put_count / 1e9),
+            finite_or_default(cur_st_get_time / cur_st_get_count / 1e9)
         );
     }
+    let elapsed = start.elapsed();
+    let elapsed_secs = elapsed.as_secs() as f64;
```
Comment (pyakushin) [inline src/bin/bobp.rs]: better to use inner float representation of the duration [as_secs_f64()](https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_secs_f64)

File: src/bin/bobp.rs [hunk 2]
```diff
@@ -1,150 +1,494 @@
...
         stat.put_total.fetch_add(1, Ordering::SeqCst);
     }
 }
 
-#[tokio::main]
-async fn main() {
-    let matches = App::new("Bob benchmark tool")
+async fn test_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    put_worker(net_conf.clone(), task_conf.clone(), stat.clone()).await;
+    get_worker(net_conf.clone(), task_conf.clone(), stat.clone()).await;
```
Comment (pyakushin) [inline src/bin/bobp.rs]: redundant clone()

File: src/bin/bobp.rs [hunk 3]
```diff
@@ -1,150 +1,494 @@
...
+}
+
+async fn ping_pong_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    let mut client = net_conf.build_client().await;
+
+    let get_options = task_conf.find_get_options();
+    let put_options = task_conf.find_put_options();
+    let measure_time = task_conf.is_time_measurement_thread();
+    let upper_idx = task_conf.low_idx + task_conf.count;
```
Comment (pyakushin) [inline src/bin/bobp.rs]: what if low_idx set to 0 in config?

Comment (idruzhitskiy) [reply src/bin/bobp.rs]: We'll iterate from 0 to count

Comment (pyakushin) [reply src/bin/bobp.rs]: I made a mistake, opposite - what if low_idx > 0 in config. There will be no time measurement threads?

Comment (idruzhitskiy) [reply src/bin/bobp.rs]: Yep, currently there will be no time measuring... I'll fix this

**Pull request merged**